python/ray/train/gbdt_trainer.py

```diff
@@ -70,9 +72,13 @@ class GBDTTrainer(BaseTrainer):
         "use_gpu",
         "placement_strategy",
     ]
+    _handles_checkpoint_freq = True
+    _handles_checkpoint_at_end = True
+
     _dmatrix_cls: type
     _ray_params_cls: type
-    _tune_callback_cls: type
+    _tune_callback_report_cls: type
+    _tune_callback_checkpoint_cls: type
```

Comment:
This is a specific pattern for some of our downstream libraries. XGBoost and LightGBM use callbacks in the framework library to save checkpoints and report to tune (i.e. the calls to `tune.checkpoint_dir()` and `tune.report()`, which will be changed to `session.report()`). 

Because LightGBM and XGBoost are so similar (LightGBM's API was based on XGBoost) we have a GBDTTrainer that can be used for most commong things. We only have to deal with a few framework-specific details, which are the actual callbacks used for saving checkpoints/reporting results, getting information from the library-native model and saving it to disk.